server/schema.ts

```diff
@@ -1,67 +1,62 @@
 import { gql } from 'https://deno.land/x/graphql_tag@0.0.1/mod.ts';
 
-
-
 export const resolvers = {
-    Query: {
-       oneRocket: async (           
-           _parent: any,
-           args: any,
-           { denostore }: any,
-           info: any
-       ) => {
-        return await denostore.cache( 
-        { info },
-        async () => {
-            const results = await fetch (
-                `https://api.spacexdata.com/v3/rockets/${args.id}`
-            ).then((res) => res.json());
-            console.log('api call');
-           
-            return results
-          }        
-        );
-       
+  Query: {
+    oneRocket: async (
+      _parent: any,
+      args: any,
+      { denostore }: any,
+      info: any
+    ) => {
+      return await denostore.cache({ info }, async () => {
+        const results = await fetch(
+          `https://api.spacexdata.com/v3/rockets/${args.id}`
+        ).then((res) => res.json());
+        // console.log('api call');
+
+        return results;
+      });
     },
-     manyRockets: async (
-        _parent: any,
-        ags: any,
-        { denostore }: any,
-        info: any 
-     ) => {
-        return await denostore.cache(
-        { info },
-         async () => {
-            const results = await fetch (
-                `https://api.spacexdata.com/v3/rockets`
-            ).then((res) => res.json());
-            console.log('api call');
-            return results
-        }
-    );
-      
-    }
-  }
-}
+    rockets: async (
+      _parent: any,
+      _args: any,
+      { denostore }: any,
+      info: any
+    ) => {
+      return await denostore.cache({ info }, async () => {
+        const results = await fetch(
+          `https://api.spacexdata.com/v3/rockets`
+        ).then((res) => res.json());
+        // console.log('api call');
+        return results;
+      });
+    },
+  },
+};
 
 export const typeDefs = gql`
-type RocketType {
-    id: String
+  type RocketType {
+    id: Int
     active: String
-    stages: String
+    stages: Int
     first_flight: String
     country: String
     wikipedia: String
     description: String
     rocket_id: String
     rocket_name: String
     rocket_type: String
-}
+    height: DimensionType
+    diameter: DimensionType
+  }
 
+  type DimensionType {
+    meters: Float
+    feet: Float
+  }
 
-type Query {
+  type Query {
     oneRocket(id: ID): RocketType
-    manyRockets: [RocketType]!
-}
-`
-
+    rockets: [RocketType]!
```

Comment:
I see that this is an array of RocketType being expected, but what is the use of the exclamation point here? It must return an array, or that the array values must not be null? I'm wondering if the exclamation point should be inside the brackets, or if there should be brackets in both spots. A query must resolve in an array (not null) and the array must contain values not null (of RocketType)?

Reply:
The exclamation point is to say that the fields value will never be null.